### DISCUSSION TITLE
Is possible to set an interval between 2 datetime/time

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#3002] [Archive]
I was wondering if there is a way to set an interval between 2 date times, dates or time only? Or we need to create 2 fields for that and take care of them?

**Comment by benhaynes** [comment]
There are a few ways to accomplish this:

1. `start_time` (eg: datetime) and `duration` (int, in seconds or minutes) as two separate fields
2. `start_time` and `end_time` as two separate fields
3. Create a single custom interface that stores data in a proprietary way, eg: `2009-01-06 12:00:00,2009-01-06 14:00:00`

**Reply by benhaynes** [reply]
**Posting the response inline.** You'd create a custom interface with two fields (any design you want), and then it would save the value in any proprietary format you want, such as the CSV version I posted above).

https://github.com/directus/next/blob/main/docs/guides/extensions/interfaces.md

**Comment by botzill** [comment]
And how can we make a single interface with 2 DateTime entries?  Basically, we need as well to have some restrictions there so that end date is > start date.

**Reply by benhaynes** [reply]
See above...

**Reply by botzill** [reply]
I see, so we need custom code for that. Do you think would be useful to have such a default field type?

**Reply by benhaynes** [reply]
You can add a feature request, but I don't think that will be added... it's very proprietary, and there's no real reason not to just do it as two fields (unless you have a good reason).

**Reply by botzill** [reply]
well, using 2 fields there is no real validation on that as users can input misleading data that we need to validate later on.

**Reply by benhaynes** [reply]
@rijkvanzanten is going to add a separate GitHub Discussion to plan out adding relative value validation and presets... in other words:

```json
{
    "start_date": {
        "_lt": "$VALUE[end_date]"
    },
    "end_date": {
        "_gt": "$VALUE[start_date]"
    }
}
```

Would that solve your specific validation needs?
